vault/client.go

```diff
@@ -171,7 +176,15 @@ func (b *BankVaultClient) removeSecret(removeSecret VaultSecret) error {
 	}
 
 	for _, secret := range secrets {
-
+		if isDirectory(secret) {
+			err := b.removeSecret(VaultSecret{
+				Path: path.Join(removeSecret.Path, secret),
```

Comment:
Just FYI: This is not a robust implementation. If this were to run on Windows (or even just running unit tests in windows), the path separator is `\` and not `/`. strings.Join or even just fmt.Sprintf with `/` would be the better solution. But that's not really relevant as long as nobody develops in windows.

Reply:
This does feel hacky yes. But no this shouldn't fail on windows. This is a secrets name returned by the vault api and vault consistently uses forward slashes (and does not run on windows).

After quickly skimming the go SDK and vault API i did not find a cleaner solution

Reply:
Isn't this just for [path/filepath](https://pkg.go.dev/path/filepath)?

> The path package should only be used for paths separated by forward slashes, such as the paths in URLs.

https://pkg.go.dev/path

Reply:
OOh wait I missunderstood you comment. Your right!  This is wrong all throughout the codebase..

Reply:
Oh! I might have mixed something here. All good! :)